cmd/describeVolume.go

```diff
@@ -58,21 +58,37 @@ func init() {
 }
 
 func describeVolumesExec(cmd *cobra.Command, args []string) error {
-	vf, err := newVolumeFormatter(cmd, args)
+	// Parse out all of the common cli volume flags
+	cvi := GetCliVolumeInputs(cmd, args)
+
+	// Create a cliVolumeOps object
+	cvOps := NewCliVolumeOps(cvi)
+
+	// Connect to px and k8s (if needed)
+	err := cvOps.Connect()
 	if err != nil {
 		return err
 	}
-	defer vf.close()
+	defer cvOps.Close()
 
-	vcf := volumeInspectFormatter{
-		volumeFormatter: *vf,
-	}
+	// Create the parser object
+	vcf := NewVolumeInspectFormatter(cvOps)
+
+	// Print the details
 	vcf.Print()
```

Comment:
Instead of having a vcf.GetError(), would `err := vcf.Print()` be easier? That way, the variable can be returned in returned instead of kept in an object.

Reply:
The problem is we use the String function. We can't return an error from the String function.
If we do away with the need to convert it to a string, we can always return an error. Let me know if that is ok.

Reply:
The more I think about it, I think it is better to remove the need for the String function. Let me update the PR with this modification

Reply:
awesome, thanks!

Reply:
Done